server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1088,6 +1228,32 @@ protected void parseCreateField(ParseContext context) throws IOException {
         }
     }
 
+    @Override
+    public PostParseExecutor getPostParseExecutor() {
+        if (this.script == null) {
+            return null;
+        }
+        return new PostParseExecutor() {
+            @Override
+            public void execute(PostParseContext context) {
+                script.executeAndEmit(
+                    context.searchLookup,
+                    context.leafReaderContext,
+                    0,
+                    v -> indexValue(context.pc, v));
```

Comment:
if indexValue is called for every field value, don't we end up calling the _field_names code block once per value rather than once per field? That looks like it's how things already work without your change though.

Reply:
Yes, it's the same for both script and non-script code paths.  It would be nice to tidy this up in a followup - maybe we should store fields that need to be added to the field_names field on the ParseContext?

Reply:
ok it's unrelated, let's look at this again in a follow-up.